/doc/tidb-cloud add tpcc 7.1 test result (#13696)
pingcap/docs: tidb-cloud/v7.1.0-performance-benchmarking-with-tpcc.md

---
title: TiDB Cloud TPC-C Performance Test Report
summary: Introduce the TPC-C performance test results for TiDB Cloud.
---

# TiDB Cloud TPC-C Performance Test Report

> This document provides the TPC-C performance test result for the TiDB Dedicated cluster with a cluster version of v7.1.0. This report can be used as a reference for the performance of TiDB Self-Hosted v7.1.0 clusters.

Reviewer: ```suggestion
This document provides the TPC-C performance test steps and results for a TiDB Dedicated cluster with the TiDB version of v7.1.0. This report can also be used as a reference for the performance of TiDB Self-Hosted v7.1.0 clusters.
```

Reviewer: TiDB version is used on UI
![image](https://github.com/pingcap/docs/assets/79440533/379c12a5-9aad-4783-90ed-dd2fa9a5b38b)

## Test overview

This test aims at showing the TPC-C performance of TiDB v7.1.0 in the Online Transactional Processing (OLTP) scenario.

## Test environment

- Cluster type: [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated)
- Cluster version: v7.1.0

> - Cloud provider: AWS

Reviewer 2: ```suggestion
- Cloud provider: AWS (us-west-2)
```

- Cluster configuration:

   | Node type | Node size | Node quantity | Node storage |
   |:----------|:----------|:----------|:----------|
   | TiDB      | 16 vCPU, 32 GiB | 2 | N/A |

>    | TiKV      | 16 vCPU, 64 GiB | 3 | 100 GiB |

Reviewer 2: ```suggestion
   | TiKV      | 16 vCPU, 64 GiB | 3 | 1000 GiB |
```

## Test steps

This section introduces how to perform the TPC-C performance test step by step.

> 1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#test-environment) requirements.

Reviewer 2: ```suggestion
1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#tidb-cluster) requirements.
```

   For more information, see [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

> 2. Connect to the newly created cluster and create a database named `tpcc`.

Reviewer 2: ```suggestion
2. On the benchmark executor, connect to the newly created cluster and create a database named `tpcc`.
```

   To connect to the cluster, see [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections.md).

   To create the `tpcc` database, execute the following SQL statement:

   ```sql
   CREATE DATABASE tpcc;
   ```

3. Load TPC-C data to the `tpcc` database.

   1. The test in this document is implemented based on [go-tpc](https://github.com/pingcap/go-tpc). You can download the test program using the following command:

      ```shell
      curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/pingcap/go-tpc/master/install.sh | sh
      ```

>    2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database:

Reviewer 2: ```suggestion
   2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database. Replace `${HOST}`, `${THREAD}`, and `${PASSWORD}` with your actual values. This document conducts three tests with `${THREAD}` value of `50`, `100`, and `200`.
```

      ```shell

>       go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error

Reviewer: Do we need to provide the range or a typical value of thread to achieve the test result?

>       go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error

Reviewer 2: ```suggestion
      go-tpc tpcc --host ${HOST} --warehouses 1000 prepare -P 4000 -D tpcc -T ${THREAD} --time 2h0m0s -p ${PASSWORD} --ignore-error
```

      ```

> 4. Before conducting the TPC-C test, collect the following statistics:

Reviewer: ```suggestion
4. To ensure that the TiDB optimizer can generate the optimal execution plan, execute the following SQL statements to collect statistics before conducting the TPC-C test:
```

   ```sql
   ANALYZE TABLE customer;
   ANALYZE TABLE district;
   ANALYZE TABLE history;
   ANALYZE TABLE item;
   ANALYZE TABLE new_order;
   ANALYZE TABLE order_line;
   ANALYZE TABLE orders;
   ANALYZE TABLE stock;
   ANALYZE TABLE warehouse;
   ```

   To accelerate the collection of statistics, execute the following SQL statements before collecting:

   ```sql
   SET tidb_build_stats_concurrency=16;
   SET tidb_distsql_scan_concurrency=16;
   SET tidb_index_serial_scan_concurrency=16;
   ```

5. Run the following `go-tpc tpcc` command to conduct stress tests on the TiDB Dedicated cluster. For each concurrency, the test takes two hours.

   ```shell
   go-tpc tpcc --host {{privatelink}} -P 4000 --warehouses 1000 run -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
   ```

> 6. Extract the tpmC data of New Order from the result.

Reviewer: ```suggestion
6. Extract the tpmC data of `new_order` from the result.
```

   TPC-C uses tpmC (transactions per minute) to measure the maximum qualified throughput (MQTh, Max Qualified Throughput). The transactions are the NewOrder transactions and the final unit of measure is the number of new orders processed per minute.

## Test result

> The TPC-C performance of v7.1.0 is as follows:

Reviewer: ```suggestion
The TPC-C performance of v7.1.0 in the [test environment] is as follows:
```

---
title: TiDB Cloud TPC-C Performance Test Report
summary: Introduce the TPC-C performance test results for TiDB Cloud.
---

# TiDB Cloud TPC-C Performance Test Report

> This document provides the TPC-C performance test steps and results for a TiDB Dedicated cluster with the TiDB version of v7.1.0. This report can also be used as a reference for the performance of TiDB Self-Hosted v7.1.0 clusters.

Reviewer 2: ```suggestion
This document provides the TPC-C performance test steps and results for a TiDB Dedicated cluster with the TiDB version of v7.1.0. This report can also be used as a reference for the performance of TiDB v7.1.0 clusters.
```

## Test overview

This test aims at showing the TPC-C performance of TiDB v7.1.0 in the Online Transactional Processing (OLTP) scenario.

## Test environment

### TiDB cluster

The test is conducted on a TiDB cluster with the following settings:

> - Cluster type: [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated)

Reviewer 2: ```suggestion
- Cluster type: [Dedicated Tier](/tidb-cloud/select-cluster-tier.md#dedicated-tier)
```

- Cluster version: v7.1.0
- Cloud provider: AWS (us-west-2)
- Cluster configuration:

   | Node type | Node size | Node quantity | Node storage |
   |:----------|:----------|:----------|:----------|
   | TiDB      | 16 vCPU, 32 GiB | 2 | N/A |
   | TiKV      | 16 vCPU, 64 GiB | 3 | 1000 GiB |

> ### Benchmark executor

Reviewer 3: Please also add the AWS region for the Benchmark executor.

Reviewer 2: Done https://github.com/pingcap/docs/pull/13696/commits/5afd391e501e127a52a415db6fad6738291c8a7d

The benchmark executor sends SQL queries to the TiDB cluster. In this test, its hardware configuration is as follows:

> - Machine type: AWS EC2

Reviewer 2: ```suggestion
- Machine type: Amazon EC2 (us-west-2)
```

- Instance type: c6a.2xlarge

## Test steps

This section introduces how to perform the TPC-C performance test step by step.

1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#tidb-cluster) requirements.

   For more information, see [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

2. On the benchmark executor, connect to the newly created cluster and create a database named `tpcc`.

   To connect to the cluster, see [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections.md).

   To create the `tpcc` database, execute the following SQL statement:

   ```sql
   CREATE DATABASE tpcc;
   ```

3. Load TPC-C data to the `tpcc` database.

   1. The test in this document is implemented based on [go-tpc](https://github.com/pingcap/go-tpc). You can download the test program using the following command:

      ```shell
      curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/pingcap/go-tpc/master/install.sh | sh
      ```

>    2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database. Replace `${HOST}`, `${THREAD}`, and `${PASSWORD}` with your actual values. This document conducts three tests with `${THREAD}` value of `50`, `100`, and `200`.

Reviewer: ```suggestion
   2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database. Replace `${HOST}`, `${THREAD}`, and `${PASSWORD}` with your actual values. This document conducts three tests with the `${THREAD}` value of `50`, `100`, and `200`.
```

   ```sql
   ANALYZE TABLE customer;
   ANALYZE TABLE district;
   ANALYZE TABLE history;
   ANALYZE TABLE item;
   ANALYZE TABLE new_order;
   ANALYZE TABLE order_line;
   ANALYZE TABLE orders;
   ANALYZE TABLE stock;
   ANALYZE TABLE warehouse;
   ```

   To accelerate the collection of statistics, execute the following SQL statements before collecting:

   ```sql
   SET tidb_build_stats_concurrency=16;
   SET tidb_distsql_scan_concurrency=16;
   SET tidb_index_serial_scan_concurrency=16;
   ```

5. Run the following `go-tpc tpcc` command to conduct stress tests on the TiDB Dedicated cluster. For each concurrency, the test takes two hours.

   ```shell
   go-tpc tpcc --host ${HOST} -P 4000 --warehouses 1000 run -D tpcc -T ${THREAD} --time 2h0m0s -p ${PASSWORD} --ignore-error
   ```

6. Extract the tpmC data of `NEW_ORDER` from the result.

   TPC-C uses tpmC (transactions per minute) to measure the maximum qualified throughput (MQTh, Max Qualified Throughput). The transactions are the NewOrder transactions and the final unit of measure is the number of new orders processed per minute.

> ## Test result

Reviewer: ```suggestion
## Test results
```

---
title: TiDB Cloud TPC-C Performance Test Report
---

# TiDB Cloud TPC-C Performance Test Report

## Test overview

This test aims at showing the TPC-C performance of TiDB v7.1.0 in the Online Transactional Processing (OLTP) scenario.

> ## Test environment (TiDB Cloud AWS)
> 
> ### Hardware configuration
> 
> | Nodes | vCPU + RAM |Storage (GiB) | Instance Count|
> |:----------|:----------| :----------| :---------- | 
> | TiKV      |16 vCPU, 64 GiB|     1000     |3|
> | TiDB      |16 vCPU, 32 GiB|     -     |2|
> | TPC-C  |8 vCPU, 16 GiB|      50    |1|
> 
> ### Software version
> 
> | Service type | Software version  |
> | :----------- | :---------------- |
> | PD           | v7.1.0  |
> | TiDB         | v7.1.0 |
> | TiKV         | v7.1.0 |
> 
> ### Parameter configuration
> 
> The TPC-C test is conducted with default parameters.
> 
> ### Prepare test data and run tpcc
> 
> 1. Deploy TiDB v7.1.0 using TiDB Cloud.
> 2. Connect the cluster and create a database named `tpcc`: 
>    
>    You could find the connection details in https://docs.pingcap.com/tidbcloud/set-up-private-endpoint-connections
> 
>    ```
>    create database tpcc;
>    ```
> 3. Use BenchmarkSQL to import the TPC-C 1000 Warehouse data: 
>    ```
>    go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare  -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
>    ```
> 4. Collect the statics before running tpcc
>    ```
>    analyze table customer; 
>    analyze table district; 
>    analyze table history; 
>    analyze table item; 
>    analyze table new_order; 
>    analyze table order_line; 
>    analyze table orders; 
>    analyze table stock; 
>    analyze table warehouse;
>    ```
>    You can run the below commands to accelerate the collection of statistics
>    ```
>    set tidb_build_stats_concurrency=16;
>    set tidb_distsql_scan_concurrency=16;
>    set tidb_index_serial_scan_concurrency=16;
>    ```
> 5. Run the command to perform stress tests on TiDB via  Private Endpoint. For each concurrency, the test takes 2 hours.
>    ```
>    go-tpc tpcc --host {{privatelink}} -P 4000 --warehouses 1000 run -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
>    ``` 
> 6. Extract the tpmC data of New Order from the result.

Reviewer 2: ```suggestion
## Test environment

- Cluster type: [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated)
- Cluster version: v7.1.0
- Cloud provider: AWS
- Cluster configuration:

   | Node type | Node size | Node quantity | Node storage |
   |:----------|:----------|:----------|:----------|
   | TiDB      | 16 vCPU, 32 GiB | 2 | N/A |
   | TiKV      | 16 vCPU, 64 GiB | 3 | 100 GiB |

## Test steps

This section introduces how to perform the TPC-C performance test step by step.

1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#test-environment) requirements.

   For more information, see [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

2. Connect to the newly created cluster and create a database named `tpcc`.

   To connect to the cluster, see [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections.md).

   To create the `tpcc` database, execute the following SQL statement:

   ```sql
   CREATE DATABASE tpcc;
   ```

3. Load TPC-C data to the `tpcc` database.

   1. The test in this document is implemented based on [go-tpc](https://github.com/pingcap/go-tpc). You can download the test program using the following command:

      ```shell
      curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/pingcap/go-tpc/master/install.sh | sh
      ```

   2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database:

      ```shell
      go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
      ```

4. Before conducting the TPC-C test, collect the following statistics:

   ```sql
   ANALYZE TABLE customer;
   ANALYZE TABLE district;
   ANALYZE TABLE history;
   ANALYZE TABLE item;
   ANALYZE TABLE new_order;
   ANALYZE TABLE order_line;
   ANALYZE TABLE orders;
   ANALYZE TABLE stock;
   ANALYZE TABLE warehouse;
   ```

   To accelerate the collection of statistics, execute the following SQL statements before collecting:

   ```sql
   SET tidb_build_stats_concurrency=16;
   SET tidb_distsql_scan_concurrency=16;
   SET tidb_index_serial_scan_concurrency=16;
   ```

5. Run the following `go-tpc tpcc` command to conduct stress tests on the TiDB Dedicated cluster. For each concurrency, the test takes two hours.

   ```shell
   go-tpc tpcc --host {{privatelink}} -P 4000 --warehouses 1000 run -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
   ```

6. Extract the tpmC data of New Order from the result.

   TPC-C uses tpmC (transactions per minute) to measure the maximum qualified throughput (MQTh, Max Qualified Throughput). The transactions are the NewOrder transactions and the final unit of measure is the number of new orders processed per minute.
```

## Test result

> The TPC-C performance of v7.1.0 is 
> 
> | Threads |  v7.1.0 tpmC |
> |:----------|:----------|
> |50|36159|
> |100|66742|
> |200|94406|

Reviewer 2: ```suggestion
The TPC-C performance of v7.1.0 is as follows:

| Threads |  v7.1.0 tpmC |
|:--------|:----------|
| 50  | 36,159 |
| 100 | 66,742 |
| 200 | 94,406 |
```

---
title: TiDB Cloud TPC-C Performance Test Report
---

# TiDB Cloud TPC-C Performance Test Report

> 

Reviewer 2: ```suggestion

This document provides the TPC-C performance test result for the TiDB Dedicated cluster with a cluster version of v7.1.0. This report can be used as a reference for the performance of TiDB Self-Hosted v7.1.0 clusters.

```

## Test overview

This test aims at showing the TPC-C performance of TiDB v7.1.0 in the Online Transactional Processing (OLTP) scenario.

## Test environment (TiDB Cloud AWS)

### Hardware configuration

| Service type | EC2 type | Instance count |
|:----------|:----------|:----------|
| PD        | 4C8G-20G |     3     |
| TiKV      |16C64G-1000G|     3     |
| TiDB      | 16C32G|     2     |
| TPC-C  | 8c16g|     1     |

### Software version

| Service type | Software version  |
| :----------- | :---------------- |
| PD           | v7.1.0  |
| TiDB         | v7.1.0 |
| TiKV         | v7.1.0 |

### Parameter configuration

#### TiKV parameter configuration

```yaml
[raft-engine]
prefill-for-recycle = true
```

### Prepare test data and run tpcc

1. Deploy TiDB v7.1.0 using TiDB Cloud.
2. Connect the cluster and create a database named `tpcc`: 

>    ![privatelink](/media/tidb-cloud/privatelink.png)

Reviewer 2: Is it necessary for users to use a [private endpoint]((https://docs.pingcap.com/tidbcloud/set-up-private-endpoint-connections)) when connecting to TiDB Cloud and executing the `go-tpc tpcc` command? Can they use the standard connection or VPC peering instead?

Reviewer 3: For performance testing, the private endpoint is recommended.

> | Service type | EC2 type | Instance count |
> |:----------|:----------|:----------|
> | PD        | 4C8G-20G |     3     |
> | TiKV      |16C64G-1000G|     3     |
> | TiDB      | 16C32G|     2     |
> | TPC-C  | 8c16g|     1     |
> 
> ### Software version
> 
> | Service type | Software version  |
> | :----------- | :---------------- |
> | PD           | v7.1.0  |
> | TiDB         | v7.1.0 |
> | TiKV         | v7.1.0 |
> 
> ### Parameter configuration
> 
> #### TiKV parameter configuration
> 
> ```yaml
> [raft-engine]
> prefill-for-recycle = true
> ```
> 
> ### Prepare test data and run tpcc
> 
> 1. Deploy TiDB v7.1.0 using TiDB Cloud.
> 2. Connect the cluster and create a database named `tpcc`: 
>    ![privatelink](/media/tidb-cloud/privatelink.png)
> 
>    ```
>    create database tpcc;
>    ```
> 3. Use BenchmarkSQL to import the TPC-C 1000 Warehouse data: 
>    ```
>    go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare  -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
>    ```
> 4. Collect the statics before running tpcc
>    ```
>    analyze table customer; 
>    analyze table district; 
>    analyze table history; 
>    analyze table item; 
>    analyze table new_order; 
>    analyze table order_line; 
>    analyze table orders; 
>    analyze table stock; 
>    analyze table warehouse;
>    ```
>    You can run the below commands to accelerate the collection of statistics
>    ```
>    set tidb_build_stats_concurrency=16;
>    set tidb_distsql_scan_concurrency=16;
>    set tidb_index_serial_scan_concurrency=16;
>    ```
> 5. Run the command to perform stress tests on TiDB via privatelink. For each concurrency, the test takes 2 hours.
>    ```
>    go-tpc tpcc --host {{privatelink}} -P 4000 --warehouses 1000 run -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
>    ``` 
> 6. Extract the tpmC data of New Order from the result.

Reviewer 2: ````suggestion
## Test environment

- Cluster type: [TiDB Dedicated](/tidb-cloud/select-cluster-tier.md#tidb-dedicated)
- Cluster version: v7.1.0
- Cloud provider: AWS
- Cluster configuration:

   | Node type | Node size | Node quantity | Node storage |
   |:----------|:----------|:----------|:----------|
   | TiDB      | 16 vCPU, 32 GiB | 2 | N/A |
   | TiKV      | 16 vCPU, 64 GiB | 3 | 100 GiB |

## Test steps

This section introduces how to perform the TPC-C performance test step by step.

1. In the [TiDB Cloud console](https://tidbcloud.com/), create a TiDB Dedicated cluster that meets the [test environment](#test-environment) requirements.

   For more information, see [Create a TiDB cluster](/tidb-cloud/create-tidb-cluster.md).

2. Connect to the newly created cluster and create a database named `tpcc`.

   To connect to the cluster, see [Connect via Private Endpoint](/tidb-cloud/set-up-private-endpoint-connections.md).

   To create the `tpcc` database, execute the following SQL statement:

   ```sql
   CREATE DATABASE tpcc;
   ```

3. Load TPC-C data to the `tpcc` database.

   1. The test in this document is implemented based on [go-tpc](https://github.com/pingcap/go-tpc). You can download the test program using the following command:

      ```shell
      curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/pingcap/go-tpc/master/install.sh | sh
      ```

   2. Run the following `go-tpc tpcc` command to import 1,000 warehouses to the `tpcc` database:

      ```shell
      go-tpc tpcc --host {{privatelink}} --warehouses 1000 prepare -P 4000 -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
      ```

4. Before conducting the TPC-C test, collect the following statistics:

   ```sql
   ANALYZE TABLE customer;
   ANALYZE TABLE district;
   ANALYZE TABLE history;
   ANALYZE TABLE item;
   ANALYZE TABLE new_order;
   ANALYZE TABLE order_line;
   ANALYZE TABLE orders;
   ANALYZE TABLE stock;
   ANALYZE TABLE warehouse;
   ```

   To accelerate the collection of statistics, execute the following SQL statements before collecting:

   ```sql
   SET tidb_build_stats_concurrency=16;
   SET tidb_distsql_scan_concurrency=16;
   SET tidb_index_serial_scan_concurrency=16;
   ```

5. Run the following `go-tpc tpcc` command to conduct stress tests on the TiDB Dedicated cluster. For each concurrency, the test takes two hours.

   ```shell
   go-tpc tpcc --host {{privatelink}} -P 4000 --warehouses 1000 run -D tpcc -T {{thread}} --time 2h0m0s -p {{password}} --ignore-error
   ```

6. Extract the tpmC data of New Order from the result.

   TPC-C uses tpmC (transactions per minute) to measure the maximum qualified throughput (MQTh, Max Qualified Throughput). The transactions are the NewOrder transactions and the final unit of measure is the number of new orders processed per minute.
````

## Test result

> The TPC-C performance of v7.1.0 is 
> 
> | Threads |  v7.1.0 tpmC |
> |:----------|:----------|
> |50|37904|
> |100|67790|
> |200|93479|
> 
> ![TPC-C](/media/tidb-cloud/v7.1.0-tpmC.png)

Reviewer 2: ```suggestion
The TPC-C performance of v7.1.0 is as follows:

| Threads |  v7.1.0 tpmC |
|:--------|:----------|
| 50  | 37,904 |
| 100 | 67,790 |
| 200 | 93,479 |

![TPC-C](/media/tidb-cloud/v7.1.0-tpmC.png)
```
